db/db.go

```diff
@@ -129,8 +135,20 @@ func (l *Loader) BeginTx(ctx context.Context, opts *sql.TxOptions) (Tx, error) {
 	return l.DB.BeginTx(ctx, opts)
 }
 
-func (l *Loader) FlushInterval() time.Duration {
-	return l.flushInterval
+func (l *Loader) BatchBlockFlushInterval() int {
+	return l.batchBlockFlushInterval
+}
+
+func (l *Loader) LiveBlockFlushInterval() int {
+	return l.liveBlockFlushInterval
+}
+
+func (l *Loader) FlushNeeded() bool {
+	totalRows := 0
+	for pair := l.entries.Oldest(); pair != nil; pair = pair.Next() {
+		totalRows += pair.Value.Len()
+	}
```

Comment:
Could we keep a running count while we insert/delete rows directly? No need to implement it, it's fine for now, just add a comment about the potential feasibility here